src/main/service/listManipulation/OpaPublicListService.ts

```diff
@@ -0,0 +1,114 @@
+import { formatDate } from '../../helpers/dateTimeHelper';
+import { OpaPressListService } from './OpaPressListService';
+
+const opaPressListService = new OpaPressListService();
+
+export class OpaPublicListService {
+    public formatOpaPublicList(opaPublicListJson: string): any {
+        const rows = [];
+        JSON.parse(opaPublicListJson).courtLists.forEach(courtList => {
+            courtList.courtHouse.courtRoom.forEach(courtRoom => {
+                courtRoom.session.forEach(session => {
+                    session.sittings.forEach(sitting => {
+                        sitting.hearing.forEach(hearing => {
+                            const allDefendants = this.processPartyRoles(hearing);
+                            allDefendants.forEach(defendant => {
+                                const caseDetails = this.buildCaseDetails(hearing.case);
+                                rows.push({
+                                    ...caseDetails,
+                                    ...defendant,
+                                });
+                            });
+                        });
+                    });
+                });
+            });
+        });
+        return rows;
+    }
+
+    private processPartyRoles(hearing): any {
+        const defendants = [];
+        hearing.party?.forEach(party => {
+            if (party.partyRole === 'DEFENDANT') {
+                const defendant = this.buildDefendantDetails(party);
+                if (defendant) {
+                    defendants.push(defendant);
+                }
+            }
+        });
+        const prosecutor = opaPressListService.processProsecutor(hearing);
+        const allDefendants = [];
+        defendants.forEach(defendant => {
+            allDefendants.push({ ...defendant, prosecutor });
+        });
+        return allDefendants;
+    }
+
+    private buildDefendantDetails(party): any {
+        if (party.individualDetails) {
+            const individual = party.individualDetails;
+            return {
+                name: this.buildIndividualName(individual),
+                offence: this.buildOffences(individual),
+            };
+        } else if (party.organisationDetails) {
+            const organisation = party.organisationDetails;
+            return {
+                name: organisation.organisationName,
+                offence: this.buildOffences(organisation),
+            };
+        }
+        return null;
+    }
+
+    private buildIndividualName(individual) {
+        const firstName = ListParseHelperService.writeStringIfValid(individual?.individualFirstName);
+        const middleNames = ListParseHelperService.writeStringIfValid(individual?.individualMiddleName);
+        const surname = ListParseHelperService.writeStringIfValid(individual?.individualSurname);
+        const forenames = [firstName, middleNames].filter(n => n.length > 0).join(' ');
+        return [surname, forenames]
```

Comment:
I am not sure if this is right for OPA public list. The example screenshots in the ticket has name in the following format "forename middleName surname" but you are returning "surname, firstname middleName". Has this been changed from the ticket?